serde/src/de/impls.rs

```diff
@@ -82,15 +141,39 @@ impl Visitor for BoolVisitor {
         Ok(v)
     }
 
+    #[cfg(feature = "std")]
+    fn visit_str<E>(&mut self, s: &str) -> Result<bool, E>
+        where E: Error,
+    {
+        match s.trim() {
+            "true" => Ok(true),
+            "false" => Ok(false),
+            _ => Err(Error::invalid_type(Type::Bool)),
+        }
+    }
+
+    #[cfg(all(not(feature = "std"), feature = "rustc_unicode"))]
     fn visit_str<E>(&mut self, s: &str) -> Result<bool, E>
         where E: Error,
     {
+        use rustc_unicode::str::UnicodeStr;
         match s.trim() {
             "true" => Ok(true),
             "false" => Ok(false),
             _ => Err(Error::invalid_type(Type::Bool)),
         }
     }
+
+    #[cfg(all(not(feature = "std"), not(feature = "rustc_unicode")))]
+    fn visit_str<E>(&mut self, s: &str) -> Result<bool, E>
```

Comment:
why not just use this version? Is there an advantage to using the std version when it's there?